x-pack/legacy/plugins/apm/public/components/app/Main/route_config/index.tsx

```diff
@@ -107,6 +110,74 @@ export const routes: BreadcrumbRoute[] = [
     ),
     name: RouteName.AGENT_CONFIGURATION
   },
+
+  {
+    exact: true,
+    path: '/settings/agent-configuration/create',
+    breadcrumb: i18n.translate(
+      'xpack.apm.breadcrumb.settings.createAgentConfigurationTitle',
+      { defaultMessage: 'Create Agent Configuration' }
+    ),
+    name: RouteName.AGENT_CONFIGURATION_CREATE,
+    component: () => {
+      const { search } = history.location;
+
+      // Ignoring here because we specifically DO NOT want to add the query params to the global route handler
+      // @ts-ignore
+      const { pageStep } = toQuery(search);
+
+      return (
+        <Settings>
+          <AgentConfigurationCreateEdit
+            pageStep={pageStep || 'choose-service-step'}
+          />
+        </Settings>
+      );
+    }
+  },
+  {
+    exact: true,
+    path: '/settings/agent-configuration/edit',
+    breadcrumb: i18n.translate(
+      'xpack.apm.breadcrumb.settings.editAgentConfigurationTitle',
+      { defaultMessage: 'Edit Agent Configuration' }
+    ),
+    name: RouteName.AGENT_CONFIGURATION_EDIT,
+    component: () => {
+      const { search } = history.location;
+
+      // typescript complains because `pageStop` does not exist in `APMQueryParams`
+      // Going forward we should move away from globally declared query params and this is a first step
+      // @ts-ignore
+      const { name, environment, pageStep } = toQuery(search);
+
+      // hooks rule is complaining about not using a hook here but it works so 🤷
+      // eslint-disable-next-line react-hooks/rules-of-hooks
+      const { data } = useFetcher(
+        callApmApi => {
+          return callApmApi({
+            method: 'POST',
+            pathname: '/api/apm/settings/agent-configuration/view',
```

Comment:
Maybe you could change it to stick what we agreed?
```suggestion
            pathname: '/api/apm/settings/agent_configuration/view',
```

Reply:
Unfortunately the API has been like this [since 7.3](https://github.com/elastic/kibana/blob/7.3/x-pack/legacy/plugins/apm/server/routes/settings.ts#L33) and this would therefore be a breaking change. So it's not easily changable until 8.0